src/js/utils/spec-helpers/ajax-helpers.js

```diff
@@ -17,204 +28,264 @@ define(['sinon', 'underscore', 'URI'], function(sinon, _, URI) {
+     * Get a reference to the mocked server, and respond to all requests
+     * with the specified response.
+     *
+     * @param {spec} spec the current spec
+     * @param {string} response the fake response.
+     * @returns {*} The current request.
      */
-    fakeRequests = function (spec) {
-        return function () {
-            var requests = [],
-                xhr = sinon.useFakeXMLHttpRequest(),
-                args = Array.prototype.slice.call(arguments);
-            xhr.onCreate = function(request) {
-                requests.push(request);
-            };
-            spec.apply(null, args.concat([requests]));
+    fakeServer = function(spec, response) {
+        var server = sinon.fakeServer.create();
+        spec.after(function() {
+            server.restore();
+        });
+        server.respondWith(response);
+        return server;
+    };
+
+    /**
+     * Keep track of all requests to a fake server, and return a reference to the Array.
+     * This allows tests to respond for individual requests.
+     *
+     * @returns {Array} An array tracking the fake requests.
+     */
+    createFakeRequests = function() {
+        var requests = [],
+            xhr = sinon.useFakeXMLHttpRequest();
+        requests.currentIndex = 0;
+        requests.restore = function() {
             xhr.restore();
         };
+        xhr.onCreate = function(request) {
+            requests.push(request);
+        };
+        return requests;
+    };
+
+    /**
+     * Keep track of all requests to a fake server, and return a reference to the Array.
+     * This allows tests to respond for individual requests.
+     *
+     * @param {test} test The test instance to create requests for.
+     * @returns {Array} An array tracking the fake requests.
+     */
+    fakeRequests = function(test) {
+        var requests = createFakeRequests();
+        test.after(function() {
+            requests.restore();
+        });
+        return requests;
+    };
+
+    /**
+     * Wraps a test function so that it is invoked with an additional parameter
+     * containing an array of fake HTTP requests.
+     *
+     * @param {function} test A function to be invoked with the fake requests.
+     * @returns {function} A wrapped version of the input function.
+     */
+    withFakeRequests = function(test) {
+        return function() {
+            var requests = createFakeRequests(),
+                args = Array.prototype.slice.call(arguments);
+            test.apply(null, args.concat([requests]));
+            requests.restore();
+        };
+    };
+
+    /**
+     * Returns the request that has not yet been responded to. If no such request
+     * is available then the current test will fail.
+     *
+     * @param {object} requests an array of fired sinon requests
+     * @returns {*} The current request.
+     */
+    currentRequest = function(requests) {
+        expect(requests.length).toBeGreaterThan(requests.currentIndex);
```

Comment:
This makes the assumption that we're testing with jasmine, or a test framework which implements `expect`. I think we can simply throw an error, which would fail the test as well.